spring-boot/proxy/src/main/java/eu/coatrack/proxy/security/ApiKeyAuthTokenVerifier.java

```diff
@@ -87,17 +65,18 @@ public Authentication authenticate(Authentication authentication) throws Authent
             if (apiKey.equals(ApiKey.API_KEY_FOR_YGG_ADMIN_TO_ACCESS_PROXIES)) {
 
                 Set<SimpleGrantedAuthority> authoritiesGrantedToYggAdmin = new HashSet<>();
-                authoritiesGrantedToYggAdmin.add(new SimpleGrantedAuthority(ServiceApiAccessRightsVoter.ACCESS_SERVICE_AUTHORITY_PREFIX + "refresh"));
+                authoritiesGrantedToYggAdmin.add(new SimpleGrantedAuthority(
+                        ServiceApiAccessRightsVoter.ACCESS_SERVICE_AUTHORITY_PREFIX + "refresh"));
 
                 ApiKeyAuthToken apiKeyAuthToken = new ApiKeyAuthToken(apiKey, authoritiesGrantedToYggAdmin);
                 apiKeyAuthToken.setAuthenticated(true);
                 return apiKeyAuthToken;
             }
 
             // regular api consumer's api key check
-            if (isApiKeyValid(apiKey)) {
+            if (apiKeyRequester.isApiKeyValid(apiKey)) {
                 // key is valid, now get service api URI identifier
-                ServiceApi serviceApi = getServiceApiByApiKey(apiKey);
+                ServiceApi serviceApi = serviceApiProvider.getServiceApiByApiKey(apiKey);
```

Comment:
Doesn't the apiKey have a field `serviceApi`?

Reply:
It has and therefore the redundant service request is now removed.